app/models/backend/BalanceRequestResponse.scala

```diff
@@ -23,12 +23,26 @@ import models.values.{BalanceId, CurrencyCode}
 import play.api.libs.json.{Json, OFormat}
 import uk.gov.hmrc.play.json.Union
 
+import java.text.NumberFormat
+import java.util.{Currency, Locale}
+
 sealed abstract class BalanceRequestResponse extends Product with Serializable
 
 case class BalanceRequestSuccess(
   balance: BigDecimal,
   currency: CurrencyCode
-) extends BalanceRequestResponse
+) extends BalanceRequestResponse {
+
+  override def toString: String =
```

Comment:
Could you possibly use a different method name? I'm just worried that, for example, the connector tests would print out confusing information if the tests fail (as it would try to format the response in that way). I think it's ok as a display type format method, but I don't know if it feels right for the default toString.